ProjectTaskTracker.js

```diff
@@ -1,77 +1,152 @@
-	const newTaskForm = document.createElement('textarea');
-	newTaskForm.classList.add("createNewTask" + toDoTaskNumber);
-	newTaskForm.classList.add("tasks");
-	newTaskForm.setAttribute("draggable", "true");
+function addNewProjectTextarea() {
+	const newProjectTextarea = document.createElement('textarea');
+	newProjectTextarea.classList.add("createNewProject" + projectNumber, "projects");
+	return newProjectTextarea
+}
 
-	
-	const toDoList = document.getElementsByClassName("toDoList")[0];
+function addToDoTask(){
+	const newTaskForm = addToDoTextarea();
 	toDoList.appendChild(newTaskForm);
 	toDoTaskNumber++;
 }
 
 function addNewProject() {
-	const newProject = document.createElement('textarea');
-	newProject.classList.add("createNewProject" + projectNumber);
-	newProject.classList.add("projects");
-	newProject.setAttribute("draggable", "true");
-
+	const newProject = addNewProjectTextarea();
 	const projectList = document.getElementsByClassName("sideBar")[0];
 	projectList.appendChild(newProject);
 	projectNumber++;
 }
 
 addTaskButton.addEventListener('click', ()=> addToDoTask());
 addNewProjectButton.addEventListener('click', ()=> addNewProject());
+/*---------------------------------------------------------------------------------------*/
+
+/*----------------------------develop draggable event------------------------------------*/
+const toDoTaskDraggable = new Draggable.Draggable(toDoList, {
+  		draggable: '.tasks'
+	});
+
+const progressTaskDraggable = new Draggable.Draggable(inProgressList, {
+  		draggable: '.tasks'
+	});
 
-// //make Project elements draggable
-// const projectElement = document.getElementsByClassName('sideBar')[0];
-// const projectDraggable =new Draggable.Draggable(projectElement, {
-// 	draggable: '.projects'
-// 	});
+const doneTaskDraggable = new Draggable.Draggable(doneList, {
+  		draggable: '.tasks'
+	});
 
-// //make tasks draggable
-// const taskElement = document.getElementsByClassName('toDoList')[0];
-// const tasksDraggable = new Draggable.Draggable(taskElement, {
-// 	draggable: '.tasks'
-// 	});
+const threeLists = [toDoList, inProgressList, doneList];
 
-const tasks = document.getElementsByClassName('tasks')[0];
-if(tasks){
-	tasks.on('drag:start', ()=> console.log('dragging'));
+let currentMousePositionX = 0;
+let currentMousePositionY = 0;
 
+let currentDraggingItem;
+let dragFromList;
+
+function getDraggingElement(event) {
+	const originalSource = event.data.originalSource;
+	const elementClassName = originalSource.className.split(" ")[0];
+	currentDraggingItem = document.getElementsByClassName(elementClassName)[0];
 }
-// if(tasks){
-// 	for(let task of tasks){
-// 		console.log("here");
-// 		task.on('drag:start', () => console.log('drag:start'));
-// 	}
-// }	
 
+function dragStart(event){
+	//find out which list the element came from and mark down which element is being dragged
+	dragFromListName = (event.sourceContainer.className).split(" ")[1];
+	for(let list of threeLists){
```

Comment:
Why not just make list of their classnames directly? e.g. `for (let listName of ['todo', 'inprogress', 'completed'])` instead of making a list of the elements and getting their classnames?

Reply:
Oh nvm i see that you use the `threeLists` again down there